src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -175,19 +176,19 @@ Copyright (c) .NET Foundation. All rights reserved.
           Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and '$(HasRuntimeOutput)' == 'true'">
 
     <!-- The following RID errors are asserts, and we don't expect them to ever occur. The error message is added as a safeguard.-->
-    <NETSdkError Condition="'$(SelfContained)' == 'true' and '$(RuntimeIdentifier)' == ''"
+    <NETSdkError Condition="'$(SelfContained)' == 'true' and '$(RuntimeIdentifier)' == '' and '$(_IsPublishing)' == 'true'"
```

Comment:
If we're handling `SelfContained` properly, I'm not sure this message makes sense here.

Reply:
May you elaborate a bit on this please? The message can occur if you turn off the RID inference by setting `RuntimeSpecific=false`. In addition, this error is acting as an assert like described in the comment above to make sure the inference is working correctly.

Reply:
OK, sounds good.